src/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -1257,7 +1267,10 @@ public void HttpWebRequest_EndGetRequestStreamContext_ExpectedValue(Uri remoteSe
             await LoopbackServer.CreateServerAsync((server, url) =>
             {
                 HttpWebRequest request = WebRequest.CreateHttp(url);
-                request.BeginGetResponse(null, null);
+                RequestState state = new RequestState();
+                state.Request = request;
+
+                request.BeginGetResponse(ResponseCallback, state);
```

Comment:
(You don't need to pass callbacks in these cases... you can just take the resulting IAsyncResult that's returned and pass it to the corresponding End method.  That'll let you control exactly when the End is invoked.  And it'll ensure that exceptions are propagated out of there rather than being stored into state.SavedException, which it appears is being ignored in many of these cases.